### PR TITLE
Add keep-to-library widget to keep card.

### PR DESCRIPTION
**Pull request by yipingliao**
@joshm1 

Adding the keep-to-library widget to the master keep button on the keep card.

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/keep/addKeep.js [hunk 1]
```diff
@@ -145,6 +145,7 @@ angular.module('kifi')
             return lib.access !== 'read_only';
           });
           scope.librarySelection = {};
+          scope.librarySelection.library = _.find(scope.libraries, { 'kind': 'system_main' });
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keep/addKeep.js:148]: This line used to be in the widget code, but it doesn't work that well that way, so I am refactoring it back out. Not directly related to keep card.

File: kifi-backend/modules/shoebox/angular/src/keep/keepToLibraryWidget.js [hunk 2]
```diff
@@ -278,12 +278,6 @@ angular.module('kifi')
 
 
         //
-        // On link.
-        //
-        scope.librarySelection.library = _.find(scope.libraries, { 'kind': 'system_main' });
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keep/keepToLibraryWidget.js:283]: Refactoring stuff.

File: kifi-backend/modules/shoebox/angular/src/keeps/keeps.js [hunk 3]
```diff
@@ -124,9 +124,6 @@ angular.module('kifi')
         // 'selection' keeps track of which keeps have been selected.
         scope.selection = new selectionService.Selection();
 
-        // Not used in template right now but will be when we hide bulk privacy changes.
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keeps/keeps.js:127]: This is just clean up. This scope variable is set elsewhere by a watch.

File: kifi-backend/modules/shoebox/angular/src/keeps/keeps.js [hunk 4]
```diff
@@ -500,6 +497,7 @@ angular.module('kifi')
                 });
 
                 scope.librarySelection = {};
+                scope.librarySelection.library = _.find(scope.libraries, { 'kind': 'system_main' });
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keeps/keeps.js:500]: Again, refactoring.

File: kifi-backend/modules/shoebox/angular/src/tagManage/tagManage.js [hunk 5]
```diff
@@ -78,6 +78,7 @@ angular.module('kifi')
             return lib.access !== 'read_only';
           });
           $scope.librarySelection = {};
+          $scope.librarySelection.library = _.find($scope.libraries, { 'kind': 'system_main' });
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/tagManage/tagManage.js:81]: Refactoring.

File: kifi-backend/modules/shoebox/angular/src/keep/keep.js [hunk 6]
```diff
@@ -900,11 +903,31 @@ angular.module('kifi')
         scope.isKeptPrivate = keep.isMyBookmark && keep.isPrivate;
         scope.isKeptPublic = keep.isMyBookmark && !keep.isPrivate;
         scope.isNotKept = !keep.isMyBookmark;
-        $log.info(keep);
 
-        scope.keepAction = function () {
-          // TODO(josh)
+        scope.librarySelection = {};
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js:907]: The widget expects the user's libraries to be on `scope.libraries`. Then, the library that the user selects in the widget is set in `scope.librarySelection`, and the widget calls `scope.clickAction`. In this case, selecting a library in the widget results in `scope.keep` being kept to the selected library.

**Comment by joshm1** [comment]
lgtm, thanks!

**Pull request merged**